src/EditorFeatures/CSharpTest/UsePatternCombinators/CSharpUsePatternCombinatorsDiagnosticAnalyzerTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.UsePatternCombinators;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UsePatternCombinators
+{
+    public class CSharpUsePatternCombinatorsDiagnosticAnalyzerTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (new CSharpUsePatternCombinatorsDiagnosticAnalyzer(), new CSharpUsePatternCombinatorsCodeFixProvider());
+
+        private static readonly string s_initialMarkup = @"
+using System;
+using System.Collections.Generic;
+class C
+{
+    bool field = {|FixAllInDocument:EXPRESSION|};
+    bool Method() => EXPRESSION;
+    bool Prop1 => EXPRESSION;
+    bool Prop2 { get } = EXPRESSION;
+    void If() { if (EXPRESSION) ; }
+    void Argument1() => Test(EXPRESSION);
+    void Argument2() => Test(() => EXPRESSION);
+    void Argument3() => Test(_ => EXPRESSION);
+    void For() { for (; EXPRESSION; ); }
+    void Local() { var local = EXPRESSION; }
+    void Assignment() { _ = EXPRESSION; }
+    void Do() { do ; while (EXPRESSION); }
+    void While() { while (EXPRESSION) ; }
+    void When() { _ = o switch { _ when EXPRESSION => 0 }; }
+    bool Return() { return EXPRESSION; }
+    IEnumerable<bool> YieldReturn() { yield return EXPRESSION; }
+    Func<object, bool> SimpleLambda() => o => EXPRESSION;
+    Func<bool> ParenthesizedLambda() => () => EXPRESSION;
+}
+";
+
+        [InlineData("i == 0")]
+        [InlineData("i > 0")]
+        [InlineData("i is C")]
+        [InlineData("i is C c")]
+        [InlineData("!(i > 0)")]
+        [InlineData("o != null")]
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsUsePatternCombinators)]
+        public async Task TestMissingOnExpression(string expression)
+        {
+            await TestMissingAsync(s_initialMarkup.Replace("EXPRESSION", expression));
+        }
+
+        [InlineData("!(o is C c)", "o is not C c")]
```

Comment:
tehnically not legal.  a designator can't appear under a `not` or `or` pattern.  We are going to discuss in LDM if we can allow this though.  I'm strongly for allowing it.

Reply:
Even if that happens within C# 9.0 timeframe, I suppose variables under `or` wouldn't? I am very hopeful that this will the camel's nose to that tent.

Reply:
i'd support you driving the 'or' case :)